platform/ios/src/MGLUserLocationAnnotationView.h

```diff
@@ -9,15 +10,15 @@ NS_ASSUME_NONNULL_BEGIN
 @class MGLUserLocation;
 
 /** View representing an MGLUserLocation on screen. */
-@interface MGLUserLocationAnnotationView : UIView
+@interface MGLUserLocationAnnotationView : MGLAnnotationView
 
 @property (nonatomic, weak) MGLMapView *mapView;
-@property (nonatomic) MGLUserLocation *annotation;
-@property (nonatomic, readonly, nullable) CALayer *haloLayer;
+@property (nonatomic, readonly, nullable) MGLUserLocation *userLocation;
 
-- (instancetype)initInMapView:(MGLMapView *)mapView NS_DESIGNATED_INITIALIZER;
-- (void)setupLayers;
+- (instancetype)initInMapView:(MGLMapView *)mapView userLocation:(MGLUserLocation *)userLocation;
```

Comment:
I know we’ve done it this way for a long time, but I’m not sure we want to make public an API that’s based on a backpointer. It’s just too tempting for subclasses to abuse this backpointer in ways that’ll leak memory. Instead, we should probably figure out a way to push the necessary state to this class.